languages.ts

```diff
@@ -48,17 +59,29 @@ const cppFilterDefinitions: HandlerArgs['filterDefinitions'] = ({
     const imports = fileContent
         .split(/\r?\n/)
         .map(line => {
-            const match = /^#include "(.*)"$/.exec(line)
-            return match ? match[1] : undefined
+            // Rewrite `@import x.y.z;` as x/y/z to simulate path matching.
+            // In plain C and C++ files, expect this to be empty.
+            const matchImport = /^@import (\w+);$/.exec(line)
+            if (matchImport) {
+                return matchImport[1].replace(/\./g, '/')
+            }
+
+            // Capture paths from #include and #import directives.
+            // N.B. Only user paths ("") are captured, not system (<>) paths.
+            const matchInclude = /^#(include|import) "(.*)"$/.exec(line)
+            return matchInclude ? matchInclude[2] : undefined
         })
         .filter((x): x is string => Boolean(x))
 
+    // Select results whose file path shares a suffix with some import.
+    // N.B. Paths are compared without file extensions.
     const filteredResults = results.filter(result => {
-        return imports.some(i =>
-            path
-                .join(path.parse(result.file).dir, path.parse(result.file).name)
-                .endsWith(path.join(path.parse(i).dir, path.parse(i).name))
-        )
+        const rp = path.parse(result.file)
```

Comment:
Would you mind giving this a written-out variable name? Maybe it's just me but I'm not sure what `rp`/`ip` are supposed to mean 🙇

Reply:
Done. Their scope is small enough and their usage obvious enough I don't think it really helps legibility any, but I am not picky about it.

Reply:
Thanks, I understand now what you meant!